sqlalchemy-stubs/sql/operators.pyi

```diff
@@ -1,111 +1,161 @@
-    def reverse_operate(self, op: Any, other: Any, **kwargs: Any) -> None: ...
+        return_type: Optional[
+            Union[Type[TypeEngine[Any]], TypeEngine[Any]]
+        ] = ...,
+    ) -> Callable[[Any], ClauseElement]: ...
+    def bool_op(
+        self, opstring: Any, precedence: int = ...
+    ) -> Callable[[Any], ClauseElement]: ...
+    def operate(
+        self, op: Any, *other: Any, **kwargs: Any
+    ) -> ClauseElement: ...
+    def reverse_operate(
+        self, op: Any, other: Any, **kwargs: Any
+    ) -> ClauseElement: ...
 
 class custom_op:
     __name__: str = ...
     opstring: Any = ...
-    precedence: Any = ...
-    is_comparison: Any = ...
-    natural_self_precedent: Any = ...
-    eager_grouping: Any = ...
-    return_type: Any = ...
+    precedence: int = ...
+    is_comparison: bool = ...
+    natural_self_precedent: bool = ...
+    eager_grouping: bool = ...
+    return_type: Optional[Union[Type[TypeEngine[Any]], TypeEngine[Any]]] = ...
     def __init__(
         self,
         opstring: Any,
         precedence: int = ...,
         is_comparison: bool = ...,
-        return_type: Optional[Any] = ...,
+        return_type: Optional[
+            Union[Type[TypeEngine[Any]], TypeEngine[Any]]
+        ] = ...,
         natural_self_precedent: bool = ...,
         eager_grouping: bool = ...,
     ) -> None: ...
-    def __eq__(self, other: Any) -> Any: ...
-    def __hash__(self) -> Any: ...
-    def __call__(self, left: Any, right: Any, **kw: Any): ...
+    def __eq__(self, other: Any) -> bool: ...
+    def __hash__(self) -> int: ...
+    def __call__(self, left: Any, right: Any, **kw: Any) -> ClauseElement: ...
 
 class ColumnOperators(Operators):
     timetuple: Any = ...
-    def __lt__(self, other: Any) -> Any: ...
-    def __le__(self, other: Any) -> Any: ...
-    __hash__: Any = ...
-    def __eq__(self, other: Any) -> Any: ...
-    def __ne__(self, other: Any) -> Any: ...
-    def is_distinct_from(self, other: Any): ...
-    def is_not_distinct_from(self, other: Any): ...
-    isnot_distinct_from: Any = ...
-    def __gt__(self, other: Any) -> Any: ...
-    def __ge__(self, other: Any) -> Any: ...
-    def __neg__(self): ...
-    def __contains__(self, other: Any): ...
-    def __getitem__(self, index: Any): ...
-    def __lshift__(self, other: Any): ...
-    def __rshift__(self, other: Any): ...
-    def concat(self, other: Any): ...
-    def like(self, other: Any, escape: Optional[Any] = ...): ...
-    def ilike(self, other: Any, escape: Optional[Any] = ...): ...
-    def in_(self, other: Any): ...
-    def not_in(self, other: Any): ...
-    notin_: Any = ...
-    def not_like(self, other: Any, escape: Optional[Any] = ...): ...
-    notlike: Any = ...
-    def not_ilike(self, other: Any, escape: Optional[Any] = ...): ...
-    notilike: Any = ...
-    def is_(self, other: Any): ...
-    def is_not(self, other: Any): ...
+    def __lt__(self, other: Any) -> ClauseElement: ...
```

Comment:
with this one, technically everything returned here would be ColumnElement or perhaps ColumnOperators.  ClauseElement would be too general, unless there are cases where that's happening.